R/FilterState.R

```diff
@@ -46,26 +46,24 @@ FilterState <- R6::R6Class( # nolint
   public = list(
 
     #' @description
-    #' Initialize a `FilterState` object
-    #' @param x (`vector`)\cr
+    #' Initialize a `FilterState` object.
+    #' @param x (`vector`)
     #'   values of the variable used in filter
-    #' @param x_reactive (`reactive`)\cr
+    #' @param x_reactive (`reactive`)
     #'   returning vector of the same type as `x`. Is used to update
     #'   counts following the change in values of the filtered dataset.
     #'   If it is set to `reactive(NULL)` then counts based on filtered
     #'   dataset are not shown.
-    #' @param slice (`teal_slice`)\cr
+    #' @param slice (`teal_slice`)
     #'   object created by [teal_slice()]
-    #' @param extract_type (`character(0)`, `character(1)`)\cr
+    #' @param extract_type (`character(0)`, `character(1)`)
```

Comment:
I'm not a fan of `character(0)` in the first place, at least in this case. We talked in another issue about changing this to `extract_type = c("none", "list", "matrix")` but it got drowned out by different topics.

I think we usually use "or" when multiple types are allowed. I'm not sure how many cases there are where it's `character` but of different lengths. How about, since more than one length is allowed, we stick to `character` and explain the allowed values in the description od `@details`?

Reply:
> How about, since more than one length is allowed, we stick to character and explain the allowed values in the description od @details?

In this case, we already have some additional information in the parameter description. I believe that this information is sufficient and do you think we still need to put it under @details tag ?

In my personal opinion, the parameter details should be self-contained. 

here is additional details from parameter decripton
<img width="482" alt="image" src="https://github.com/insightsengineering/teal.slice/assets/6700955/f4c5a594-d1ee-4b65-a587-484c4b8eed6d">

Reply:
Yes, this description is self-contained :+1: